source/php-frameworks/symfony.txt

```diff
@@ -0,0 +1,397 @@
+install the ``twig`` bundle:
+
+.. code-block::
+
+   composer require symfony/twig-bundle
+
+.. note::
+
+   This step might result in an error message about unset environment
+   variables, but this is resolved in the following section.
+
+Modify Project Files
+~~~~~~~~~~~~~~~~~~~~
+
+This section demonstrates how to modify the files in your
+``restaurants`` project to create a Symfony web application that views
+restaurants based on filter criteria.
+
+Set Environment Variables
+`````````````````````````
+
+In the root directory, navigate to the ``.env`` file and define the
+following environment variables or edit them if they are already present:
+
+.. code-block:: none
+   :caption: .env
+
+   ...
+
+   MONGODB_URL=<your Atlas connection string>
+   MONGODB_DB=sample_restaurants
+
+To retrieve your connection string, see the
+:ref:`php-symfony-atlas-cluster` step.
+
+Create Restaurant Entity and Controller
+```````````````````````````````````````
+
+Create the ``Restaurant.php`` file in the ``src/Document`` directory and
+paste the following code to create an entity that represents documents in
+the ``restaurants`` collection:
+
+.. literalinclude:: /includes/php-frameworks/Restaurant.php
+   :caption: src/Document/Restaurant.php
+   :language: php
+
+Next, create the ``RestaurantController.php`` file in the
+``src/Controller`` directory and paste the following code to handle the
+endpoints in your application:
+
+.. literalinclude:: /includes/php-frameworks/RestaurantController.php
+   :caption: src/Controller/RestaurantController.php
+   :language: php
+
+The controller file defines the ``index()`` method, which displays text on
+the web app's front page. The file also defines the ``browse()`` method,
+which finds documents in which the ``borough``
+field is ``'Queens'`` and the ``name`` field contains the string ``'Moon'``.
+This method then displays the documents at the ``/restaurant/browse/`` route. The
+``browse()`` method uses the ``QueryBuilder`` class to construct the query.
+
+Specify Routes
+``````````````
+
+In the ``config/routes.yaml`` file, add the following routes:
+
+.. code-block:: yaml
+   :caption: config/routes.yaml
+
+   # main routes
+   restaurant_index:
+     path: /
+     controller: App\Controller\RestaurantController::index
+     methods: [GET]
+   
+   # browse restaurants
+   restaurant_browse:
+     path: /restaurant/browse
+     controller: App\Controller\RestaurantController::browse
+     methods: [GET]
+
+This file sets the following routes in the application:
+
+- ``restaurant_index [GET]``: landing page
+- ``restaurant_browse [GET]``: retrieves and displays specific restaurants
+
+Customize Templates
+```````````````````
+
+Next, create templates to customize the web app's appearance.
+
+Create the ``templates/restaurant`` directory and populate it with the
+following files:
+
+- ``index.html.twig``
+- ``browse.html.twig``
+
+Paste the following code into the ``index.html.twig`` file:
```

Comment:
Why are the Twig templates hard-coded here instead of referencing files in `includes/php-frameworks`?

Reply:
Will move